### PR TITLE
[Doc] New language model on GS tutorial

### PR DESCRIPTION
**Pull request by zhjwy9343**
*Issue #, if available:*

*Description of changes:*
This PR revise the LM + GNN tutorial, providing systematic descriptions of how to use language models plus GNN in GraphStorm.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/source/advanced/language-models.rst [hunk 1]
```diff
@@ -1,45 +1,67 @@
 .. _language_models:
 
-Use Text as Node Features
-=============================
-Many real world graphs have text contents as nodes' features, e.g., the title and description of a product, and the questions and comments from users. To leverage these text contents, GraphStorm supports language models (LMs), i.e., HuggingFace BERT models, to embed text contents and use these embeddings in Graph models' training and inference.
+Use Language Models in GraphStorm
+==================================
+Many real world graphs have text contents as nodes' features, e.g., the title and description of a product, and the questions and comments from users. To leverage these text contents, GraphStorm supports language models (LMs), i.e., HuggingFace BERT models, to embed text contents and use these embeddings for various Graph Machine Learning tasks.
```
Comment (zheng-da) [inline docs/source/advanced/language-models.rst:5]:
```suggestion
Many real world graphs have text contents as nodes' features, e.g., the title and description of a product, and the questions and comments from users. To leverage these text contents, GraphStorm supports multiple language models (LMs) from HuggingFace to embed text contents and use these embeddings for various Graph Machine Learning tasks.
```

Comment (zhjwy9343) [reply docs/source/advanced/language-models.rst:5]: changed.

File: docs/source/advanced/language-models.rst [hunk 2]
```diff
@@ -1,45 +1,67 @@
...
+==================================
+Many real world graphs have text contents as nodes' features, e.g., the title and description of a product, and the questions and comments from users. To leverage these text contents, GraphStorm supports multiple language models (LMs) from HuggingFace to embed text contents and use these embeddings for various Graph Machine Learning tasks.
 
-There are two modes of using LMs in GraphStorm:
+There are a set of modes to use LMs in GraphStorm.
 
-* Embed text contents with pre-trained LMs, and then use them as the input node features, without fine-tuning the LMs. Training speed in this mode is fast, and memory consumption will be lower. However, in some cases, pre-trained LMs may not fit to the graph data well, and fail to improve performance.
+#. Use Pre-trained LMs only without fine-tuning
+    In this mode, users can embed text contents with pre-trained LMs, and then use them as the input node features to train GNN models, but do not fine-tune the LMs. Model training speed in this mode is fast, and memory consumption will be lower. However, in some cases, pre-trained LMs may not fit to the graph data well, hence not improving performance.
```
Comment (wangz10) [inline docs/source/advanced/language-models.rst:10]: Suggest changing "pre-trained LMs may not fit to the graph data well" to "pre-trained LMs may not fit to the text data on graph"

File: docs/source/advanced/language-models.rst [hunk 3]
```diff
@@ -1,45 +1,67 @@
...
+
+    * Co-train LMs and GNN
 
-    All commands below should be run in a GraphStorm Docker container. Please refer to the :ref:`GraphStorm Docker environment setup<setup>` to prepare your environment.
+    In this mode, LMs and GML models are co-train in the same training epoch simultaneousely, which will better fit the LMs and GNN models to graph data. However, co-train LMs and GNN models will consume much more memory, particularly GPU memory, and take much longer time to complete training loops.
 
-    If you :ref:`set up the GraphStorm environment with pip Packages<setup_pip>`, please replace all occurrences of "2222" in the argument ``--ssh-port`` with **22**, and clone GraphStorm toolkits.
+This tutorial will help users to learn how to use GraphStorm for all of the above modes. Given that using language models on text could take longer time than general GNN only training, this tutorial uses a relatively small demo graph for a quick-start. Users can refer the `GraphStorm MAG example <https://github.com/awslabs/graphstorm/tree/main/examples/mag>`_ for an large texture graph example.
```
Comment (wangz10) [inline docs/source/advanced/language-models.rst:31]: "texture" -> "textual"

File: docs/source/advanced/language-models.rst [hunk 4]
```diff
@@ -150,20 +249,206 @@ To co-train BERT and GNN models, we need to add one more argument, ``--lm-train-
...
+1. Well pre-trained LM and GNN before the GLEM co-training: empirically, LM or GNN models that are not well-trained lead to degraded performance when co-training with GLEM directly. Therefore, users need to pre-train the LMs and GNN Models first.
+
+To pre-train LMs and GNN models, users can follow the :ref:`22two_step_mannually` instruction and save the best LMs and GNN models.
+
+2. The pseudolabeling technique: this technique predicts pseudolabels on the unlabeled nodes and uses them as additional supervision signals for mutual distillation between LM and GNN. This can be enabled by the setting the ``use_pseudolabel`` argument.
+
+GraphStorm configurations
+##########################
+
+To use GLEM, users need to set a new configuration se, called ``training_method``, which specifies how to utilize specific model training method. Users can refer to the ``acm_glem_nc_pretrain.yaml`` that includes the following ``training_method`` related configurations.
```
Comment (wangz10) [inline docs/source/advanced/language-models.rst:370]: "configuration se"?

File: docs/source/advanced/language-models.rst [hunk 5]
```diff
@@ -150,20 +249,206 @@ To co-train BERT and GNN models, we need to add one more argument, ``--lm-train-
...
+    training_method:
+      name: glem
+      kwargs:
+        em_order_gnn_first: false
+        inference_using_gnn: true
+        pl_weight: 0.5
+        num_pretrain_epochs: 100
+    use_pseudolabel: true
+
+Within the ``traing_method`` section, there are two important configurations. First, the ``pl_weight`` defines the weights of pseudolabel, which determines the importance of pseudolabel. Users can lower the value to reduce the influence of using pseudolabel. The second important configuration is the ``num_pretrain_epochs``. The GLEM method provides its own pre-training implementation, which train LMs and GNN models iteratively in one epoch, i.e., first fix GNN model and train LMs in one forward and backward loop, and then fix LM but use it to embed text as input for GNN models to be trained in one loop. In the pre-training epochs, GLEM will not use the pseudolabel, but the true labels only, even if users set the ``use_pseudolabel`` configuration to be true.
```
Comment (wangz10) [inline docs/source/advanced/language-models.rst:383]: "importance of pseudolabel" -> importance of pseudolabel in mutual distillation

File: examples/use_your_own_data/acm_glem_nc_pretrain.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    fanout: "50,50"
+    num_layers: 2
+    hidden_size: 256
+    use_mini_batch_infer: false
+    training_method:
+      name: glem
+      kwargs:
+        em_order_gnn_first: false
+        inference_using_gnn: true
+        pl_weight: 0.5
```
Comment (wangz10) [inline examples/use_your_own_data/acm_glem_nc_pretrain.yaml:31]: Have you tried a lower value (e.g. 0.1)?

**Review by wangz10 — CHANGES_REQUESTED** [review]

File: docs/source/advanced/language-models.rst (hunk 2 above)
Comment (zhjwy9343) [reply docs/source/advanced/language-models.rst:10]: thanks, changed.

File: docs/source/advanced/language-models.rst (hunk 3 above)
Comment (zhjwy9343) [reply docs/source/advanced/language-models.rst:31]: thanks. changed.

File: docs/source/advanced/language-models.rst (hunk 5 above)
Comment (zhjwy9343) [reply docs/source/advanced/language-models.rst:383]: changed.

File: docs/source/advanced/language-models.rst (hunk 4 above)
Comment (zhjwy9343) [reply docs/source/advanced/language-models.rst:370]: changed.

File: examples/use_your_own_data/acm_glem_nc_pretrain.yaml (hunk 6 above)
Comment (zhjwy9343) [reply examples/use_your_own_data/acm_glem_nc_pretrain.yaml:31]: Yes, I tried 0.1. The performance was not improved, or even worse in some settings.

**Review by wangz10 — APPROVED** [review]

**Pull request merged**